src/js/upcoming-this-month.js

```diff
@@ -0,0 +1,97 @@
+import TmdbApi from './tmdb-api.js';
+
+const tmdb = new TmdbApi();
+
+async function getUpcomingMovies() {
+  try {
+    const movies = await tmdb.getUpcomingMovies();
+    const today = new Date();
+    const thisMonthMovies = movies.filter(movie => {
+      const releaseDate = new Date(movie.release_date);
+      return (
+        releaseDate.getFullYear() === today.getFullYear() &&
+        releaseDate.getMonth() === today.getMonth()
+      );
+    });
+
+    if (thisMonthMovies.length === 0) {
+      displayMessage('No upcoming movies this month.');
+    } else {
+      const randomMovie =
+        thisMonthMovies[Math.floor(Math.random() * thisMonthMovies.length)];
+      displayMovie(randomMovie);
+    }
+  } catch (error) {
+    console.error('Failed to fetch upcoming movies:', error);
+  }
+}
+
+function displayMessage(message) {
+  const movieContainer = document.getElementById('movie-container');
+  movieContainer.innerHTML = `<p>${message}</p>`;
+}
+
+function displayMovie(movie) {
+  const movieContainer = document.getElementById('movie-container');
+  const imageUrl = `https://image.tmdb.org/t/p/original/${movie.backdrop_path}`;
+  const releaseDate = new Date(movie.release_date).toLocaleDateString();
+  const genres = movie.genre_ids.map(id => genreMap[id]).join(', ');
+
+  const html = `
+    <div class="upcoming-container" >
+      <div class="upcoming-img">
+        <img class="upcoming-img" src="${imageUrl}" alt="${movie.title}"></div>
+        <div class="movie-details">
+            <h2 class="movie-title">${movie.title}</h2>
+            <p class="detail-item">Release date: <span class="relase-date">${releaseDate}</span></p>
+            <p class="detail-item">Vote / Votes: <span class="vote-count">${movie.vote_count}</span></p>
+            <p class="detail-item">Popularity: <span class="popularity-value">${movie.popularity}</span></p>
+            <p class="genres-item">Genre: <span class="genres">${genres}</span></p>
+            <p class="about">ABOUT</p>
+            <p class="overview">${movie.overview}</p>
+            <button  id="library-btn" data-id="${movie.id}">Add to my library</button>
+        </div>
+    </div>
+    `;
+
+  movieContainer.innerHTML = html;
+
+  const libraryBtn = document.getElementById('library-btn');
```

Comment:
library już napisałem, mogłaś użyć metod, rzuć okiem do local-movie-manager.js
najlepiej jakbyś pod to zmieniła kod

Reply:
Chodzi o ten kod konkretnie  z local-movie-manager.js, tak?
addMovie(movie) {
    const index = this.movies.findIndex(ele => ele.id === movie.id);
    if (index === -1) {
      this.movies.push(movie);
      this._saveData();
    } else {
      console.log('Movie already added:', movie);
    }
  }

  removeMovie(id) {
    const index = this.movies.findIndex(ele => ele.id === id);
    if (index !== -1) {
      this.movies.splice(index, 1);
      this._saveData();
    } else {
      console.log('Movie not found');
    }
  }

  getMovies() {
    return this.movies;
  }
}

Reply:
Dokładnie tak
użyj tych metod tam gdzie trzeba

tylko musisz najpierw import tej klasy zrobić do siebie
tak samo jak z klasą api robiłaś
i stworzy instancje